### DISCUSSION TITLE
rpc call yields empty resultset

### DISCUSSION THREAD

**Discussion by korve** [supabase/supabase#4643] [Questions] [answered]
I've created a function that returns results that are close to a lat/lng point on earth:

```
CREATE OR REPLACE FUNCTION find_nearest_offers(latitude float, longitude float)
  RETURNS TABLE (id bigint, title varchar, description text, address varchar, price real)
  LANGUAGE plpgsql AS
$func$
BEGIN
   RETURN QUERY
   SELECT o.id, o.title, o.description, o.address, o.price FROM public.offers o ORDER BY o.location <-> ST_GeomFromText (CONCAT('Point(', latitude, ' ', longitude, ')'), 4326);
END
$func$;
```

When i call the function directly from the SQL Editor i get the expected a results. But when i call the function using a RPC i get an empty array. Here is the JS Code:

```
await supabase
		.rpc("find_nearest_offers", {
			latitude,
			longitude
		})
```

Can someone point me to a possible cause of this problem?

**Comment by korve** [comment, marked as answer]
Ok... For anyone else having the same problem. "Role based security" was enabled which prohibited access to the data. After disabling RLS i got the expected response.

**Reply by kungpaogao** [reply]
I spent way too long on this, thank you!

**Comment by zackdotcomputer** [comment]
For anyone else coming here looking for an answer to this - you can also add an RLS policy that allows the user to view all the rows that are serving as input to your function. If the given user could query all the rows and perform the work themselves locally, it appears the RLS logic allows them to access the RPC and get the data base.
